apiserver/facades/controller/crossmodelrelations/crossmodelrelations.go

```diff
@@ -447,10 +450,106 @@ func (api *CrossModelRelationsAPIv3) registerOneRemoteRelation(
+	ctx context.Context, remoteRelationArgs params.RemoteEntityArgs,
+) (params.RemoteRelationWatchResults, error) {
+	results := params.RemoteRelationWatchResults{
+		Results: make([]params.RemoteRelationWatchResult, len(remoteRelationArgs.Args)),
+	}
+
+	for i, arg := range remoteRelationArgs.Args {
+		w, changes, err := api.watchOneRelationChanges(ctx, arg)
+		if err != nil {
+			results.Results[i].Error = apiservererrors.ServerError(err)
+			continue
+		}
+
+		// Ignore the changes, this method and the watcher return
+		// different structures.
+		watcherID, _, err := internal.EnsureRegisterWatcher(ctx, api.watcherRegistry, w)
+		if err != nil {
+			results.Results[i].Error = apiservererrors.ServerError(err)
+			continue
+		}
+
+		results.Results[i].RemoteRelationWatcherId = watcherID
+		results.Results[i].Changes = changes
+	}
+
+	return results, nil
+}
+
+func (api *CrossModelRelationsAPIv3) watchOneRelationChanges(
+	ctx context.Context,
+	arg params.RemoteEntityArg,
+) (common.RelationUnitsWatcher, params.RemoteRelationChangeEvent, error) {
+	var empty params.RemoteRelationChangeEvent
+	// relationToken is the relation UUID.
+	relationToken := arg.Token
+
+	relationKey, err := api.relationService.GetRelationKeyByUUID(ctx, arg.Token)
+	if err != nil {
+		return nil, empty, internalerrors.Errorf("getting relation key for %q: %w", arg.Token, err)
+	}
+	relationTag := names.NewRelationTag(relationKey.String())
+
+	relationUUID := corerelation.UUID(relationToken)
+	if err := api.checkMacaroonsForRelation(ctx, relationUUID, relationTag, arg.Macaroons, arg.BakeryVersion); err != nil {
+		return nil, empty, internalerrors.Capture(err)
+	}
+
+	appToken, err := api.crossModelRelationService.GetOfferingApplicationToken(ctx, relationUUID)
+	if err != nil {
+		return nil, empty, internalerrors.Errorf("getting offering relation tokens: %w", err)
+	}
+	w, err := api.relationService.WatchRelationUnits(ctx, relationUUID, appToken)
+	if err != nil {
+		return nil, empty, internalerrors.Errorf("watching relation units: %w", err)
+	}
+
+	change, err := api.getRemoteRelationChangeEvent(ctx, relationUUID, appToken)
+	if err != nil {
+		return nil, empty, internalerrors.Errorf("getting initial change: %w", err)
+	}
+
+	wrapped, err := wrappedRelationChangesWatcher(w, appToken, relationUUID, api.relationService)
+	if err != nil {
+		return nil, empty, internalerrors.Errorf("watching relation changes: %w", err)
+	}
+
+	return wrapped, change, nil
+}
+
+func (api *CrossModelRelationsAPIv3) getRemoteRelationChangeEvent(
+	ctx context.Context,
+	relationUUID corerelation.UUID,
+	applicationUUID coreapplication.UUID,
+) (params.RemoteRelationChangeEvent, error) {
+	change, err := api.relationService.GetFullRelationUnitChange(ctx, relationUUID, applicationUUID)
+	if err != nil {
+		return params.RemoteRelationChangeEvent{}, err
+	}
+
+	appSetings := transform.Map(change.ApplicationSettings, func(k string, v string) (string, interface{}) { return k, v })
```

Comment:
This will create an empty map if the application settings is nil. We want to preserve a nil map if it is nil. A test for this case should be added.